src/comm/Tcp.cc

```diff
@@ -18,19 +19,16 @@
 #if HAVE_NETINET_IN_H
 #include <netinet/in.h>
 #endif
-#if HAVE_SYS_SOCKET_H
-#include <sys/socket.h>
-#endif
 #include <type_traits>
 
-/// setsockopt(2) wrapper
+/// xsetsockopt(2) wrapper
 template <typename Option>
 static bool
 SetSocketOption(const int fd, const int level, const int optName, const Option &optValue)
 {
     static_assert(std::is_trivially_copyable<Option>::value, "setsockopt() expects POD-like options");
     static_assert(!std::is_same<Option, bool>::value, "setsockopt() uses int to represent boolean options");
-    if (setsockopt(fd, level, optName, reinterpret_cast<const char *>(&optValue), sizeof(optValue)) < 0) {
+    if (xsetsockopt(fd, level, optName, reinterpret_cast<const char *>(&optValue), sizeof(optValue)) < 0) {
```

Comment:
Please remove this cast if it is no longer needed:

```suggestion
    if (xsetsockopt(fd, level, optName, &optValue, sizeof(optValue)) < 0) {
```

Reply:
removed